src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -115,207 +90,123 @@ private void OnTransactionAvailable(TransactionReceived transactionReceived)
-                        // While the wallet is catching up the entire node will wait.
-                        // If a wallet is recovered to a date in the past. Consensus will stop until the wallet is up to date.
-
-                        // TODO: This code should be replaced with a different approach
-                        // Similar to BlockStore the wallet should be standalone and not depend on consensus.
-                        // The block should be put in a queue and pushed to the wallet in an async way.
-                        // If the wallet is behind it will just read blocks from store (or download in case of a pruned node).
-
-                        next = newTip.GetAncestor(next.Height + 1);
-                        Block nextblock = null;
-                        int index = 0;
-                        while (true)
-                        {
-                            if (cancellationToken.IsCancellationRequested)
-                            {
-                                this.logger.LogTrace("(-)[CANCELLATION_REQUESTED]");
-                                return;
-                            }
-
-                            nextblock = this.blockStore.GetBlock(next.HashBlock);
-                            if (nextblock == null)
-                            {
-                                // The idea in this abandoning of the loop is to release consensus to push the block.
-                                // That will make the block available in the next push from consensus.
-                                index++;
-                                if (index > 10)
-                                {
-                                    this.logger.LogTrace("(-)[WALLET_CATCHUP_INDEX_MAX]");
-                                    return;
-                                }
-
-                                // Really ugly hack to let store catch up.
-                                // This will block the entire consensus pulling.
-                                this.logger.LogWarning("Wallet is behind the best chain and the next block is not found in store.");
-                                Thread.Sleep(100);
-                                continue;
-                            }
-
-                            break;
-                        }
-
-                        this.walletTip = next;
-                        this.walletManager.ProcessBlock(nextblock, next);
-                    }
-                }
-                else
-                {
-                    ChainedHeader findTip = this.walletTip.FindAncestorOrSelf(newTip);
-                    if (findTip == null)
-                    {
-                        this.logger.LogTrace("(-)[NEW_TIP_BEHIND_NOT_IN_WALLET]");
-                        return;
-                    }
-
-                    this.logger.LogDebug("Wallet tip '{0}' is ahead or equal to the new tip '{1}'.", this.walletTip, newTip);
-                }
+                Thread.Yield();
```

Comment:
why this instead of
`this.walletSynchronisationLoop?.Dispose();`

Reply:
I agree. Changed.